esi_leap/api/controllers/v1/offer.py

```diff
@@ -66,15 +67,8 @@ def get_one(self, offer_id):
         cdict = request.to_policy_values()
         policy.authorize('esi_leap:offer:get', cdict, cdict)
 
-        o_id = offer.Offer.get(offer_id)
-
-        if len(o_id) > 1:
-            raise exception.OfferDuplicateName(name=offer_id)
-        elif len(o_id) == 0:
-            raise exception.OfferNotFound(offer_uuid=offer_id)
-
-        o = OffersController._add_offer_availabilities(o_id[0])
-
+        o_object = utils.get_authorized_offer(offer_id, None, cdict)
```

Comment:
I kinda suspect that this should be `get_offer_by_uuid_or_name_contract_creation` (or, if you rename that function as I suggest, simply `get_offer`.

Reply:
fixed